### PR TITLE
OAM-186: Added home facility validation for ward/service type

### PR DESCRIPTION
**Pull request by sradziszewski**
JIRA task link: https://openlmis.atlassian.net/browse/OAM-186

### CODE REVIEW

File: src/test/java/org/openlmis/referencedata/validate/UserValidatorTest.java [hunk 1]
```diff
@@ -255,6 +268,18 @@ public void shouldRejectIfUserHasNoRightForEditAndInvariantsWereChanged() {
     assertErrorMessage(errors, ROLE_ASSIGNMENTS, UserMessageKeys.ERROR_FIELD_IS_INVARIANT);
   }
 
+  @Test
+  public void shouldRejectIfHomeFacilityHasWardOrServiceType() {
```
Comment (tsznaj) [inline src/test/java/org/openlmis/referencedata/validate/UserValidatorTest.java:272]: Please also add a test where facilityRepository.findById returns empty Optional.

**Review by tsznaj — CHANGES_REQUESTED** [review]
Please check one comment.

File: src/test/java/org/openlmis/referencedata/validate/UserValidatorTest.java (hunk 1 above)
Comment (sradziszewski) [reply src/test/java/org/openlmis/referencedata/validate/UserValidatorTest.java:276]: I added extra test for that case.

**Review by tsznaj — APPROVED** [review]
Approved!

**Pull request merged**